test/integration/provider/admin/backend_apis_controller_test.rb

```diff
@@ -53,4 +53,21 @@ class Provider::Admin::BackendApisControllerTest < ActionDispatch::IntegrationTe
     assert_response :redirect
     assert_equal 'https://new-endpoint.com:443/p', backend_api.reload.private_endpoint
   end
+
+  test 'delete a backend api with products' do
+    backend_api = @provider.backend_apis[0]
+    assert backend_api.backend_api_configs.any?
+
+    delete provider_admin_backend_api_path(backend_api)
+    assert_nothing_raised(ActiveRecord::RecordNotFound) { backend_api.reload }
+  end
+    
+  test 'delete a backend api without any products' do
+    backend_api = @provider.backend_apis[1]
```

Comment:
If you do https://github.com/3scale/porta/pull/1131/files#r319517769, then this one should be either `backend_api = @provider.backend_apis.second` or `backend_api = @provider.backend_apis.last` to use the same everywhere in the test 😄

Reply:
@duduribeiro @Martouta 
I personally don't like that syntax for this particular case. I prefer using them in the logic where it can have a syntactical meaning. Here it's just an index that coincidentally can be referred as first/second/last but that could change anytime.

Reply:
That's fine for me 👍 😄 Specially because it has the assertion later of ww you need anyway (I meant the `assert/assert_not backend_api.backend_api_configs.any?`)